src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2227,15 +2331,36 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 value = intrinsic_procedures.comptime_eval(call_name, al, x.base.base.loc, args);
             }
             tmp = ASR::make_FunctionCall_t(al, x.base.base.loc, stemp,
-                nullptr, args.p, args.size(), nullptr, 0, a_type, value, nullptr);
+                s_generic, args.p, args.size(), nullptr, 0, a_type, value, nullptr);
         } else if(ASR::is_a<ASR::Subroutine_t>(*s)) {
             tmp = ASR::make_SubroutineCall_t(al, x.base.base.loc, stemp,
-                nullptr, args.p, args.size(), nullptr);
+                s_generic, args.p, args.size(), nullptr);
         } else {
             throw SemanticError("Unsupported call type for " + call_name,
                 x.base.base.loc);
         }
     }
+    int select_generic_procedure(const Vec<ASR::expr_t*> &args,
+            const ASR::GenericProcedure_t &p, Location loc) {
+        for (size_t i=0; i < p.n_procs; i++) {
+
+            if( ASR::is_a<ASR::ClassProcedure_t>(*p.m_procs[i]) ) {
+                ASR::ClassProcedure_t *clss_fn
```

Comment:
@certik  As discussed, we should add these helper functions in `asr_utils` right?

Reply:
Yes.

Reply:
No, simply modify files in the `libasr` subdirectory as part of this PR.